src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -350,6 +359,38 @@
         (update-in-nil ["containment_path"]
                        anonymize-containment-path newcontext config))))
 
+(pls/defn-validated anonymize-metric :- metric-schema-str
+  [metric :- metric-schema-str
+   context
+   config]
+  (-> metric
+      (update-in ["name"] anonymize-leaf :name context config)
```

Comment:
Sometimes the 'name' presents a resource type ... this should follow the same anonymization rules as resource type elsewhere to keep the shape the same.